components/omega/test/ocn/StateTest.cpp

```diff
@@ -0,0 +1,343 @@
+
+int initStateTest() {
+
+   int Err = 0;
+
+   // Initialize the Machine Environment class - this also creates
+   // the default MachEnv. Then retrieve the default environment and
+   // some needed data members.
+   OMEGA::MachEnv::init(MPI_COMM_WORLD);
+   OMEGA::MachEnv *DefEnv = OMEGA::MachEnv::getDefaultEnv();
+   MPI_Comm DefComm       = DefEnv->getComm();
+
+   // Initialize the IO system
+   Err = OMEGA::IO::init(DefComm);
+   if (Err != 0)
+      LOG_ERROR("State: error initializing parallel IO");
+
+   // Create the default decomposition (initializes the decomposition)
+   Err = OMEGA::Decomp::init();
+   if (Err != 0)
+      LOG_ERROR("State: error initializing default decomposition");
+
+   // Initialize the default halo
+   Err = OMEGA::Halo::init();
+   if (Err != 0)
+      LOG_ERROR("State: error initializing default halo");
+
+   // Initialize the default mesh
+   Err = OMEGA::HorzMesh::init();
+   if (Err != 0)
+      LOG_ERROR("State: error initializing default mesh");
+
+   return Err;
+}
+
+//------------------------------------------------------------------------------
+// The test driver for State -> This tests the time level update of state
+// variables and verifies the state is read in correctly.
+//
+int main(int argc, char *argv[]) {
+
+   int RetVal = 0;
+
+   // Initialize the global MPI environment
+   MPI_Init(&argc, &argv);
+   Kokkos::initialize();
+   {
+
+      // Call initialization routine to create the default decomposition
+      int Err = initStateTest();
+      if (Err != 0)
+         LOG_CRITICAL("State: Error initializing");
+
+      // Get MPI vars if needed
+      OMEGA::MachEnv *DefEnv = OMEGA::MachEnv::getDefaultEnv();
+      MPI_Comm Comm          = DefEnv->getComm();
+      OMEGA::I4 MyTask       = DefEnv->getMyTask();
+      OMEGA::I4 NumTasks     = DefEnv->getNumTasks();
+      bool IsMaster          = DefEnv->isMasterTask();
+
+      OMEGA::HorzMesh *DefHorzMesh = OMEGA::HorzMesh::getDefault();
+      OMEGA::Decomp *DefDecomp     = OMEGA::Decomp::getDefault();
+      OMEGA::Halo *DefHalo         = OMEGA::Halo::getDefault();
+
+      // These hard-wired variables need to be upated
+      // with retrivals/config options
+      int NVertLevels = 60;
+      int NTimeLevels = 2;
+
+      // Create meta data dimensions
+      OMEGA::MetaDim::create("NCells", DefHorzMesh->NCellsSize);
+      OMEGA::MetaDim::create("NEdges", DefHorzMesh->NEdgesSize);
+      OMEGA::MetaDim::create("NVertLevels", NVertLevels);
+
+      for (int NTimeLevels = 2; NTimeLevels < 4; NTimeLevels++) {
+
+         int CurLevel = NTimeLevels - 2;
+         int NewLevel = CurLevel - 1;
+         if (NewLevel < 0) {
+            NewLevel = NTimeLevels - 1;
+         }
+
+         // Create "default" state
+         if (NTimeLevels == 2) {
+
+            OMEGA::OceanState::init();
+            OMEGA::OceanState *DefOceanState = OMEGA::OceanState::getDefault();
```

Comment:
```suggestion

```

Reply:
@mwarusz, I think we still need to retrieve a pointer following the `init` call similar to what we do for `MachEnv`, `Decomp`, `Halo` etc. Otherwise, I'm not sure how we'd have access to a `DefOceanState` instance.

Reply:
I looked at the whole code again and I now think that this line should stay, since `OceanState::getDefault` should be tested and is not called anywhere else.